OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/DiscussionTopicsAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package org.edx.mobile.view.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.view.ViewCompat
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import org.edx.mobile.R
+import org.edx.mobile.databinding.RowDiscussionTopicBinding
+import org.edx.mobile.interfaces.ItemsComparator
+import org.edx.mobile.interfaces.OnItemClickListener
+import org.edx.mobile.model.discussion.DiscussionTopicDepth
+import org.edx.mobile.util.UiUtils.setTextViewDrawableStart
+
+class DiscussionTopicsAdapter(
+    private val listener: OnItemClickListener<DiscussionTopicDepth>
+) : ListAdapter<DiscussionTopicDepth, DiscussionTopicsAdapter.DiscussionTopicViewHolder>(object :
+    ItemsComparator<DiscussionTopicDepth>() {
+    override fun areContentsTheSame(
+        oldItem: DiscussionTopicDepth,
+        newItem: DiscussionTopicDepth
+    ): Boolean = oldItem.discussionTopic.hasSameId(newItem.discussionTopic)
+}) {
+    private var childPadding: Int = 0
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): DiscussionTopicViewHolder {
+        val binding =
+            RowDiscussionTopicBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        childPadding = parent.context.resources.getDimensionPixelOffset(R.dimen.edx_margin);
+        return DiscussionTopicViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(
+        holder: DiscussionTopicViewHolder,
+        position: Int
+    ) {
```

Comment:
Organize the function. The method callings inside are very random. Ypu could also move the logic to a bind function inside viewholder class.